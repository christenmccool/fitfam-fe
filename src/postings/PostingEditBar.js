import React from 'react';
import { Link as RouterLink } from 'react-router-dom';

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';


/** Menu bar to allow editing a workout posting
 *
 * PostingList -> PostingEditBar
 */
const PostingEditBar = ({ postId }) => {
  return (
    <Box sx={{display:"flex", justifyContent: "flex-end"}} >
      <Button
        variant="contained"
        component={RouterLink}
        to={`/postings/${postId}/edit`}
        color="grey"
        sx={{p: 0, color: "text.secondary"}}
      >
        Edit
      </Button>
  </Box>
  )
}

export default PostingEditBar;