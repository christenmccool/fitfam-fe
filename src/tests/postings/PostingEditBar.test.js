import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import PostingEditBar from "../../postings/PostingEditBar";


it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <PostingEditBar 
        postId={1}
        isUserWo={true}
      />
    </MemoryRouter>
  );
});


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <PostingEditBar 
        postId={1}
        isUserWo={true}
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


it("renders edit button if user workout", async function () {
  render(
    <MemoryRouter>
      <PostingEditBar 
        postId={1}
        isUserWo={true}
      />
    </MemoryRouter>
  );

  expect(screen.queryByRole('link', {name: /edit/i})).toBeInTheDocument();
});


it("renders delete button if not user workout", async function () {
  render(
    <MemoryRouter>
      <PostingEditBar 
        postId={1}
        isUserWo={false}
      />
    </MemoryRouter>
  );
  expect(screen.queryByRole('link', {name: /edit/i})).not.toBeInTheDocument();
  expect(screen.queryByRole('button', {name: /delete/i})).toBeInTheDocument();
});







