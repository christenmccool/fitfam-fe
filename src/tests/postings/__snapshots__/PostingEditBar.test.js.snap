// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1bvc4cc"
  >
    <a
      class="MuiButton-root MuiButton-contained MuiButton-containedGrey MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1tcl1r1-MuiButtonBase-root-MuiButton-root"
      href="/postings/1/edit"
      tabindex="0"
    >
      Edit
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
